packages/google_maps_flutter/google_maps_flutter_web/lib/src/dom_window_extension.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:js_interop';
+
+import 'package:web/web.dart';
+
+/// This extension type exists to handle unsupported features by certain browsers.
+@JS()
+extension type WindowWithTrustedTypes(Window _) implements JSObject {
+  /// Get the `trustedTypes` object from the window, if it is supported.
+  @JS('trustedTypes')
+  external TrustedTypePolicyFactory? get trustedTypesNullable;
+}
+
+// TODO(ditman): remove this extension type when we depend on package:web 0.5.1
+/// This extension exists as a stop gap until `package:web 0.5.1` is released.
+/// That version provides the `TrustedTypes` API.
+@JS()
+extension type TrustedTypePolicyFactory._(JSObject _) implements JSObject {
```

Comment:
@srujzs if the `extension type` has the same name as the one in `package:web`, would we get an error because of the duplicate Dart name? Should this be named something else that is not an exact copy of the one defined in `package:web`?

Reply:
Just for the sake of name collisions, I prefixed these stubs with `GoogleMaps`. These will go away in the short term anyway.

Reply:
You'll see an error just because of the collision in Dart names, so if there's code that's importing this and `package:web` and uses the symbol, you'll see a collision. Namespacing the import or renaming via the `@JS` annotation are both valid alternatives.